CODING_STANDARDS.md

```diff
@@ -0,0 +1,378 @@
+Good:
+
+```tsx
+const obj = { A: 1, b: 2, C: 3 };
+
+const obj = {
+    A: 1,
+    b: 2,
+    C: 3,
+};
+
+type Type = {
+    requiredTypeOne: string;
+    requiredTypeTwo: number;
+    optionalTypeOne?: string;
+    optionalTypeTwo?: number;
+};
+```
+
+Bad:
+```tsx
+const obj = {A:1,b:2,C:3};
+
+const obj = {A:1, b:2, C:3};
+
+const obj = {A : 1, b : 2, C : 3};
+
+const obj = { "A" : 1, "b" : 2, "C" : 3 };
+
+const obj = { A : 1, b : 2, C : 3 };
+
+const obj = { A :1, b :2, C :3 };
+
+const obj = { A : 1 , b : 2 , C : 3 };
+
+const obj = {
+    A : 1,
+    b : 2,
+    C : 3,
+};
+
+
+const obj = {
+    A : 1
+  , b : 2
+  , C : 3
+};
+
+type Type = {
+    optionalTypeOne?: string;
+    requiredTypeOne: string;
+    optionalTypeTwo?: number;
+    requiredTypeTwo: number;
+};
+```
+
+#### __Arrays__
+
+- Arrays should always be created by `[]` and not `new Array()`.
+- When creating an empty Array, use `[]` without spaces.
+- When creating an Array with values
+   - The comma should have no space before and 1 space behind itself
+   - The last value shouldn't be trailed with a comma on a single line but should on a multiline
+   - The comma shouldn't start a new line. It should always trail the previous value
+
+Good:
+```tsx
+const arr = [1, 2, 3];
+
+const arr = [
+    1,
+    2,
+    3,
+];
+```
+
+Bad:
+```tsx
+const arr = [1,2,3];
+
+const arr = [ 1, 2, 3 ];
+
+const arr = [ 1 , 2 , 3 ];
+
+const arr = [
+    1, 
+    2, 
+    3
+];
+
+const arr = [
+    1
+  , 2
+  , 3 
+];
+```
+
+### __If statements__
+
+- Avoid single line `if` statements when possible
+- Opening parenthesis should not have a space after it but should have a space between it and the `if` syntax
+- Closing parenthesis should not have a space before it but should have a space after it and before the open bracket `{` syntax
+
+
+### __JSX__
+
+Always use `()` surrounding the components in the return so the tags are not offset when auto-formatting the indentations
+
+### __Blocks/if statements__
+
+- The opening brackets should be followed by 1 new line
+- The closing brackets should not be in the same line as body logic and should be followed by 1 new line
+- The opening brackets should always follow a space and not start at a new line
+- The if keyword should always be followed with a space, an opening parenthesis, and another space.
+- The test should end with a space, a closing parenthesis, and another space.
+- The if statement should always contain a block.
+- If there is an else statement, it should be on the same line as the closing bracket of the block.
+- The else statement is followed by another block and should be separated from both blocks with a single space on both sides.
+- Assignment in a test is not use.
+
+Good:
+```tsx
+if (someConst === 'some string') {
+    return true;
+}
+```
+
+Bad:
+```tsx
+if (someConst === 'some string') {
+    return false;}
+
+if (someConst === 'some string') { return false;
+}
+
+if (someConst === 'some string') return false;
```

Comment:
This is perfectly readable and has less LoC than what you specified as "good". As long as there's semicolon at the end I feel like this should be fine to use.

Reply:
I personally believe it's not as clear to read something like this `if (ref) innerRef.current = ref;`

Reply:
for most simple cases i find the first approach easier to read but i'm fine with always having curly braces
```js
    if (x === 1) return true;
    if (x === 2) return false;
    if (x === 3) return false;

    if (x === 1) {
      return true;
    }
    if (x === 2) {
      return false;
    }
    if (x === 3) {
      return true;
    }
```

Reply:
I too prefer inline returns for readability's sake when they're simple, but do agree that at a certain breakpoint/level of complexity they need brackets

Reply:
![IMG_0731](https://media3.giphy.com/media/HyDAt79JnlY4M/giphy.gif)

Reply:
at times they can be simple but the complexity isn't really quantifiable so what is readable/not for each person and will lead to code that reads differently which is the opposite of the purpose of a standardization